### DISCUSSION TITLE
Support for DC measurement

### DISCUSSION THREAD

**Discussion by gz-jgale** [David00/rpi-power-monitor#50] [Q&A]
@David00 

I found an old kit (v1.2 w/ 4x 100A Clamps and 2x 30A clamps) at work and am wondering how much work it would be to change the code and use as a DC power meter for 0-15V DC? Does it already work for DC? ( the docs seem to all talk about AC and power factors. )

**Comment by David00** [comment]
No way, cool!  So, using it for DC would be a little bit quite a challenge, but not impossible. There are two challenges.


DC current sensors work differently than the AC current sensors, and they usually require a power source of their own, which my PCB isn't set up to provide.  Instead of using a DC current sensor, you could use a shunt in series with the DC load to calculate the current, but this would take up two channels on the PCB (one to measure the voltage before the shunt, and one to measure the voltage after the shunt). Plus, you'd have to reduce these voltages prior to connecting them to the board's input so they do not exceed 1.65V at peak voltage.

The second issue is that all of the input channels are buffered with a +1.65V flat signal so that the troughs in an AC waveform can be captured.  So, in the total input range from 0 - 3.3V, an AC signal oscillates around the +1.65V mid-line.  Yet if a 0V DC signal comes into any of the inputs, my board will automatically add 1.65V, leaving only 1.65V remaining for the actual signal (since 1.65 + 1.65 = 3.3V). This effectively reduces the input sensitivity by half, and with a 10-bit ADC, that would leave you with only 512 unique values.  You could easily swap the ADC out with the MCP3208, which is a 12-bit ADC, and it would give you 2048 unique values (considering that the first 2048 values are taken by the +1.65V buffer voltage), but then you'd have to rewrite the SPI transfer algorithm to work with 12-bits instead of 10.

So, both challenges require some external circuitry to use with my PCB, and at that point, it'd probably be better to make a new PCB altogether with everything contained in one. I am interested in looking into this more in the future.
